src/main/java/Duke/Duke.java

```diff
@@ -0,0 +1,123 @@
+        greet();
+
+        Scanner in = new Scanner(System.in);
+        String userInput = "";
+        boolean isInvalid = true;
+
+        while (isInvalid) {
+            userInput = in .nextLine();
+            if (!userInput.equals("list")) {
+                isInvalid = false;
+            } else {
+                printSeparationLine();
+                System.out.println("You do not have any task to do.");
+                printSeparationLine();
+            }
+        }
+
+        while (true) {
+            if (userInput.startsWith("todo ")) {
+                Task newTask = new Todo(userInput.substring(5));
+                addTask(newTask);
+            } else if (userInput.startsWith("deadline ")) {
+                Task newTask = new Deadline(userInput.substring(9, userInput.indexOf("/")),
+                        userInput.substring(userInput.indexOf("/by") + 4));
+                addTask(newTask);
+            } else if (userInput.startsWith("event ")) {
+                Task newTask = new Event(userInput.substring(6, userInput.indexOf("/")),
+                        userInput.substring(userInput.indexOf("/at") + 4));
+                addTask(newTask);
+            } else if (userInput.startsWith("done ")) {
+                int selectedTask = Integer.parseInt(userInput.substring(5)) - 1;
+                tasks[selectedTask].setAsDone();
+                printSeparationLine();
+                System.out.println("     Nice! I've marked this task as done:");
+                System.out.println("       [" + tasks[selectedTask].getStatusIcon() + "] " +
+                        tasks[selectedTask].getDescription());
+                printSeparationLine();
+            } else if (userInput.equals("bye")) {
+                exit();
+                System.exit(0);
+            } else if (userInput.equals("list")) {
+                printSeparationLine();
+                if (numberOfTasks == 1) {
+                    System.out.println("     Here is the task in your list:");
+                } else {
+                    System.out.println("     Here are the tasks in your list:");
+                }
+                for (int i = 0; i < numberOfTasks; i++) {
+                    System.out.println("     " + (i + 1) + "." + tasks[i]);
+                }
+                printSeparationLine();
+                System.out.println();
+            }
+            userInput = in .nextLine();
+        }
+    }
+
+
+    static void greet() {
+        System.out.println(GREET_MESSAGE);
+    }
+
+
+    static void printSeparationLine() {
+        System.out.println(SEPARATION_LINE);
+    }
+
+
+    public static void addTask(Task t) {
+        tasks[numberOfTasks] = t;
+        numberOfTasks++;
+        echo(t);
+    }
+
+
+    static void echo(Task newTask) {
+        printSeparationLine();
+        System.out.println(SPACES + "Got it. I've added this task:");
+
+        System.out.println(SPACES + newTask);
+
+        if (numberOfTasks == 1) {
```

Comment:
Good Job. I like how you split the print out from main.

Reply:
Thank you for your compliment! :)

Reply:
Thank you for taking your time to review my PR, Bowen! :)